src/vs/workbench/parts/extensions/electron-browser/extensionTipsService.ts

```diff
@@ -164,41 +186,75 @@ export class ExtensionTipsService extends Disposable implements IExtensionTipsSe
 	}
 
 	getWorkspaceRecommendations(): TPromise<string[]> {
-		if (!this.isEnabled()) {
-			return TPromise.as([]);
-		}
-		const workspace = this.contextService.getWorkspace();
-		return TPromise.join([this.resolveWorkspaceRecommendations(workspace), ...workspace.folders.map(workspaceFolder => this.resolveWorkspaceFolderRecommendations(workspaceFolder))])
-			.then(recommendations => {
-				this._allWorkspaceRecommendedExtensions = distinct(flatten(recommendations));
+		if (!this.isEnabled) { return TPromise.as([]); }
+
+		return this.fetchCombinedExtensionRecommendationConfig()
+			.then(content => {
+				this._workspaceIgnoredRecommendations = content.unwantedRecommendations;
+				this._allIgnoredRecommendations = distinct([...this._globallyIgnoredRecommendations, ...this._workspaceIgnoredRecommendations]);
+				this._allWorkspaceRecommendedExtensions = content.recommendations;
+				this.refilterAllRecommendations();
 				return this._allWorkspaceRecommendedExtensions;
 			});
 	}
 
-	private resolveWorkspaceRecommendations(workspace: IWorkspace): TPromise<string[]> {
-		if (workspace.configuration) {
-			return this.fileService.resolveContent(workspace.configuration)
-				.then(content => this.processWorkspaceRecommendations(json.parse(content.value, [])['extensions']), err => []);
+	private fetchCombinedExtensionRecommendationConfig(): TPromise<IExtensionsConfigContent> {
+		const workspace = this.contextService.getWorkspace();
+		return TPromise.join([this.resolveWorkspaceExtensionConfig(workspace), ...workspace.folders.map(workspaceFolder => this.resolveWorkspaceFolderExtensionConfig(workspaceFolder))])
+			.then(contents => this.processConfigContent(this.mergeExtensionRecommendationConfigs(flatten(contents))));
```

Comment:
Looks like now, this is the only place from where we call `mergeExtensionRecommendationConfigs`, in which case, how about merging its code into `processConfigContent`?